src/android/Mapbox.java

```diff
@@ -168,7 +183,23 @@ public void run() {
             FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(webViewWidth - left - right, webViewHeight - top - bottom);
             params.setMargins(left, top, right, bottom);
             mapView.setLayoutParams(params);
-
+			//@anothar we have to override back button or app will just close
+            mapView.setOnKeyListener(new View.OnKeyListener() {
```

Comment:
@anothar I know those are separate Views and the others rather than webview do not know JS code. What I meant with "redirect" was to implement a cordova exec init callback on the plugin listening for "back" keys and emit the "back button" on the JS.